### PR TITLE
Feat :added Search and address Management

### PR DESCRIPTION
**Pull request by poornima-alugubelly**
User can search by category and product name
User add, delete and edit address
Note : For now even with empty values the form will be submitted input validation will be done soon

### CODE REVIEW

File: src/context/UserDataContext.js [hunk 1]
```diff
@@ -58,6 +60,23 @@ const UserDataProvider = ({ children }) => {
 					console.log("error", err);
 				}
 			})();
+		auth.isAuth &&
+			(async () => {
+				setLoading(true);
+				try {
+					const res = await getAddressListService(auth.token);
+					console.log("addresses", res);
```
Comment (gopalbbharadva) [inline src/context/UserDataContext.js:68]: Remove consoles, (removing consoles after functionality works is almost all we forgot.)

**Review by gopalbbharadva — APPROVED** [review]
Clean code, Functionality is also working.

File: src/backend/controllers/CartController.js [hunk 2]
```diff
@@ -146,6 +146,7 @@ export const removeItemFromCartHandler = function (schema, request) {
 console.log("in backend");
 export const updateCartItemHandler = function (schema, request) {
 	const productId = request.params.productId;
+	console.log("resquest", request);
 	const userId = requiresAuth.call(this, request);
 	try {
 		if (!userId) {
```
Comment (BelideNikhil) [inline src/backend/controllers/CartController.js:152]: please remove console logs

File: src/backend/controllers/AddressController.js [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+		);
+	}
+	const userAddressList = schema.users.findBy({ _id: userId }).addressList;
+
+	return new Response(200, {}, { addressList: userAddressList });
+};
+
+export const addAddressHandler = function (schema, request) {
+	const userId = requiresAuth.call(this, request);
+	console.log("id", userId);
```
Comment (bipin7yadav) [inline src/backend/controllers/AddressController.js:23]: console.log can be removed

**Review by bipin7yadav — APPROVED** [review]
remove consoles otherwise good to go

**Review by BelideNikhil — APPROVED** [review]
The code is really clean and well structured as always!
Great work.
Also the UI looks 🔥

**Pull request merged**